app/src/main/java/com/tari/android/wallet/model/TariWalletAddress.kt

```diff
@@ -32,75 +32,47 @@
-        } else {
-            this.hexString = hexString
-            this.emojiId = emojiId
-        }
-    }
-
-    fun isZeros(): Boolean = hexString == zeroHex || hexString == zero66Hex || hexString.all { it == '0' }
+    fun isZeros(): Boolean = hexString == HEX_ZERO || hexString == HEX_ZERO_66 || hexString.all { it == '0' }
 
     override fun equals(other: Any?): Boolean = (other is TariWalletAddress) && hexString == other.hexString
 
     override fun hashCode(): Int = hexString.hashCode()
 
     override fun toString(): String = "TariWalletAddress(hexString='$hexString', emojiId='$emojiId')"
 
-    // region Parcelable
-
-    constructor(parcel: Parcel) : this() {
-        readFromParcel(parcel)
-    }
-
-    companion object CREATOR : Parcelable.Creator<TariWalletAddress> {
-
-        const val zeroHex = "0000000000000000000000000000000000000000000000000000000000000026"
-        const val zero66Hex = "000000000000000000000000000000000000000000000000000000000000000026"
-
-        override fun createFromParcel(parcel: Parcel): TariWalletAddress {
-            return TariWalletAddress(parcel)
-        }
-
-        override fun newArray(size: Int): Array<TariWalletAddress> {
-            return Array(size) { TariWalletAddress() }
+    companion object {
+        const val HEX_ZERO = "0000000000000000000000000000000000000000000000000000000000000026"
+        const val HEX_ZERO_66 = "000000000000000000000000000000000000000000000000000000000000000026"
+        const val EMOJI_ZERO =
+            "\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF00\uD83C\uDF57"
+
+        fun createWalletAddress(hexString: String = HEX_ZERO, emojiId: String = ""): TariWalletAddress {
+            // crunch fix for not crashing on action related to wallet address
+            return if (hexString == HEX_ZERO) {
+                TariWalletAddress(
+                    hexString = HEX_ZERO_66,
+                    emojiId = EMOJI_ZERO,
+                )
+            } else {
+                TariWalletAddress(
+                    hexString = hexString,
+                    emojiId = emojiId,
+                )
+            }
         }
 
         fun validate(addressHex: String): Boolean = addressHex.length > 64
```

Comment:
Valid tari address is exactly 66 characters long (64 chars of public key plus 2 bytes of metadata/control sum).